### ISSUE TITLE
@aws-cdk\aws-amplify-alpha: branch 

### ISSUE THREAD

**Issue by PetroLiubynskyi** [aws/aws-cdk#28289]
### Describe the bug

When running stack with the following code (Python) it is not possible to deploy to AWS Amplify from asset as it is described in [documentation](https://docs.aws.amazon.com/cdk/api/v2/python/aws_cdk.aws_amplify_alpha/README.html#deploying-assets)

The code

```python
amplify_app = amplify.App(
            self, 
            "Amplify",
            app_name="my-app",
            platform=amplify.Platform.WEB
        )

deployed_bundle = assets.Asset(
            self,
            "DeployedBundle",
            path=bundle_path)

amplify_app.add_branch(
            "AmplifyBranch",
            branch_name=env,
            asset=deployed_bundle,
            auto_build=False,
            pull_request_preview=False
        )
```

The reason is reference in [branch.ts](https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/aws-amplify-alpha/lib/branch.ts) on line 236
```ts
entry: path.join(__dirname, '..', '..', 'custom-resource-handlers', 'dist', 'aws-amplify-alpha', 'asset-deployment-handler', 'index.js'),
```

There is no dependency from **@aws-cdk/custom-resource-handlers** package inside **@aws-cdk/aws-amplify-alpha**

### Expected Behavior

Amplify branch should be created from an asset

### Current Behavior

Amplify branch is not created from an asset 

The following error appears

![image](https://github.com/aws/aws-cdk/assets/124598808/5b570f98-416c-4d54-ab35-4adbe458ffc7)

### Reproduction Steps

Execute the code

```python
amplify_app = amplify.App(
            self, 
            "Amplify",
            app_name="my-app",
            platform=amplify.Platform.WEB
        )

deployed_bundle = assets.Asset(
            self,
            "DeployedBundle",
            path=bundle_path)

amplify_app.add_branch(
            "AmplifyBranch",
            branch_name=env,
            asset=deployed_bundle,
            auto_build=False,
            pull_request_preview=False
        )
```

### Possible Solution

Add dependency to **@aws-cdk/custom-resource-handlers** package inside **@aws-cdk/aws-amplify-alpha** and publish @aws-cdk/custom-resource-handlers itself

### CDK CLI Version

aws-cdk.aws-amplify-alpha==2.114.1a0

### Node.js Version

18.14.0

### OS

Windows 11 Pro

### Language

Python

### Language Version

3.10.11

**Comment by pahud** [comment]
Thanks for the report. Looks like we should add the dependency.

**Comment by PetroLiubynskyi** [comment]
The issue is not fixed in version 2.118.0a0

There is still an error when running "amplify_app.add_branch"

" Cannot find entry file at C:\Users\user1\AppData\Local\Temp\jsii-kernel-1N88ev\node_modules\@aws-cdk\custom-resource-handlers\dist\aws-amplify-alpha\asset-deployment-handler\index.js"
